Content.Client/Stains/StainsSystem.cs

```diff
@@ -0,0 +1,41 @@
+using Content.Client.Items.Systems;
+using Content.Shared.Clothing;
+using Content.Shared.Stains;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Stains;
+
+public sealed class StainsSystem : SharedStainsSystem
+{
+    [Dependency] private readonly ItemSystem _item = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<StainableComponent, AfterAutoHandleStateEvent>(OnAfterState);
+        SubscribeLocalEvent<StainableComponent, EquipmentVisualsUpdatedEvent>(OnVisualsUpdated);
+    }
+
+    private void OnAfterState(EntityUid uid, StainableComponent component, ref AfterAutoHandleStateEvent @event)
+    {
+        if (!TryComp<SpriteComponent>(uid, out var sprite))
+        {
+            return;
+        }
+        sprite.Color = component.StainColor;
```

Comment:
i dunno how feasible it is but for blood and stuff just tinting red is boring, an overlay masked with the base sprite would look way better

Reply:
we decided to make it this way for now

Reply:
ok hopefully someone will improve it down the line